apps/oxlint/src-js/bindings.d.ts

```diff
@@ -8,14 +8,19 @@ export type JsLintFileCb =
 export type JsLoadPluginCb =
   ((arg: string) => Promise<string>)
 
+/** JS callback to set settings. */
+export type JsSetSettingsCb =
+  ((arg: Record<string, any>) => void)
```

Comment:
this should just transfer a string from rust to JS, then call json.parse on the JS side. Transfering objects is more expensive thatn just transfering strings, so it should help keep performance fast.